common/include/mapf/common/socket.h

```diff
@@ -27,7 +27,7 @@ class Socket {
     Socket();
     virtual ~Socket();
 
-    int Connect(const std::string &addr, bool retry = true);
+    int Connect(const std::string &addr, int retry = -1);
```

Comment:
D'oh, this should of course also be changed:
```suggestion
    int Connect(const std::string &addr, unsigned retry = 0);
```
So 0 (default) is to retry indefinitely.

@LiorAmram, I'll apply this suggestion too, so you can rebase, test, and force-push.

Reply:
Note that I also wrote a commit message in 85e3831 - I think this should be the commit message of the squashed commit.